src/lib/server/RequestHandler.js

```diff
@@ -159,15 +159,31 @@ export function prepareOutput(req, {Index, store, getRoutes, fileName}, renderPr
   // script tag that hooks up the client side react code.
   const currentState = store.getState();
 
+  let headContent, bodyContent;
+  const renderMethod = config.server && config.server.renderMethod;
+  if (renderMethod) {
+    try {
+      const renderOutput  = await Promise.resolve(renderMethod(reactRenderFunc, main));
```

Comment:
Cool trick here… `Promise.resolve` is awesome. `await Promise.resolve(somethingHere)` when `somethingHere` is a promise, it will wait for that promise to resolve. If `somethingHere` is anything else, it will be ready immediately. This is a nice and clean way to accept a promise or a synchronous value